### PR TITLE
Leverage init container to support non-root scenario

### LINKED ISSUE #167: [FEATURES] Leverage init container to support non-root scenario
**What feature you'd like to add:**

Leverage init container to support non-root scenario

**Why is this feature needed:**

当前的问题：

1.底层存储需要指定的non root身份只读访问；
2.用户身份在ldap中配置，在主机的/etc/passwd不存在；容器很难无缝感知该用户id和用户名
3.而alluxio需要依赖用户名称，而不是用户id

### PR DESCRIPTION
**Pull request by xieydd**
### Ⅰ. Describe what this PR does
Leverage init container to support non-root scenario, Not Test Yet
### Ⅱ. Does this pull request fix one issue?
 fixes #167 
### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
Add Unit Test for function getPasswdPath getGroupPath getCreateArgs

### Ⅳ. Describe how to verify it
Later

### Ⅴ. Special notes for reviews
@cheyang

### CODE REVIEW

File: charts/alluxio/values.yaml [hunk 1]
```diff
@@ -20,6 +20,10 @@ image: alluxio/alluxio
 imageTag: 2.4.0-SNAPSHOT
 imagePullPolicy: IfNotPresent
 
+# Create User Docker Image
+createImage: xieydd/centos
```
Comment (cheyang) [inline charts/alluxio/values.yaml:24]: user creator?

File: pkg/ddc/alluxio/utils.go [hunk 2]
```diff
@@ -134,3 +138,32 @@ func (e *AlluxioEngine) isFluidNativeScheme(mountPoint string) bool {
 func (e *AlluxioEngine) getLocalStorageDirectory() string {
 	return "/underFSStorage"
 }
+
+func (e *AlluxioEngine) getPasswdPath() string {
+	timestamp := time.Now().Format("20060102150405")
```
Comment (cheyang) [inline pkg/ddc/alluxio/utils.go:143]: user Path和group path的时间戳可以保持一致。另外是否可以考虑/tmp/fluid_202009150718/作为父文件夹

File: pkg/ddc/alluxio/utils.go [hunk 3]
```diff
@@ -134,3 +138,32 @@ func (e *AlluxioEngine) isFluidNativeScheme(mountPoint string) bool {
...
+	passwd := "/tmp/passwd_" + timestamp
+	logrus.Infoln("User told me to generate passwd file in ", passwd)
```
Comment (cheyang) [inline pkg/ddc/alluxio/utils.go:145]: 这里使用的日志系统不是logrus, kubebuilder
内置的日志系统是"github.com/go-logr/logr"

**Review by cheyang — CHANGES_REQUESTED** [review]

File: Dockerfile.centos [hunk 4]
```diff
@@ -0,0 +1,3 @@
+FROM centos:latest
```
Comment (cheyang) [inline Dockerfile.centos:1]: 建议使用alpine:3.10.2

File: go.mod [hunk 5]
```diff
@@ -14,6 +14,7 @@ require (
 	github.com/onsi/ginkgo v1.11.0
 	github.com/onsi/gomega v1.8.1
 	github.com/pkg/errors v0.8.1
+	github.com/sirupsen/logrus v1.4.2
```
Comment (cheyang) [inline go.mod:17]: 使用已有日志系统，谢谢。

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -e
+
+function printUsage {
```
Comment (cheyang) [inline entrypoint.sh:4]: 是否可以考虑环境变量，而不是参数？只是建议哈，可以借鉴https://github.com/Alluxio/alluxio/blob/master/integration/docker/entrypoint.sh#L148-L163

File: pkg/ddc/alluxio/types.go [hunk 7]
```diff
@@ -68,9 +68,12 @@ type ImageInfo struct {
 }
 
 type UserInfo struct {
-	User    int `yaml:"user"`
-	Group   int `yaml:"group"`
-	FSGroup int `yaml:"fsGroup"`
+	User       int    `yaml:"user"`
+	Group      int    `yaml:"group"`
+	FSGroup    int    `yaml:"fsGroup"`
+	passwdPath string `yaml:"passwdpath"`
```
Comment (cheyang) [inline pkg/ddc/alluxio/types.go:74]: 变量名passwdPath，groupPath应该是大写，否则反射会失败。

**Review by cheyang — CHANGES_REQUESTED** [review]

File: charts/alluxio/values.yaml (hunk 1 above)
Comment (cheyang) [reply charts/alluxio/values.yaml:24]: 是否可以考虑
```
initUsers:
   image: xieydd/centos
  uid: 1005
  gid: 1005
```

Comment (xieydd) [reply charts/alluxio/values.yaml]: Done

File: pkg/ddc/alluxio/utils.go (hunk 3 above)
Comment (xieydd) [reply pkg/ddc/alluxio/utils.go]: Done

File: go.mod (hunk 5 above)
Comment (xieydd) [reply go.mod]: Done

File: pkg/ddc/alluxio/types.go (hunk 7 above)
Comment (xieydd) [reply pkg/ddc/alluxio/types.go]: Done

**Pull request merged**